### PR TITLE
Fixed Xcode project

### PR DESCRIPTION
**Pull request by hrubymar10**
This PR include:

- fixed `unit_tests` target in Xcode project
- fixed project itself

Xcode project required OpenSSL 1.1 before but libraries path in project targeted OpenSSL 1.0 which caused version mismatch. I rework it to OpenSSL 1.1 so libcrypto.dylib and OpenSSL headers in mac compile stuff must be updated to make it work.

This project works for me but I asked @singalen to test this project before merging.

### CODE REVIEW

**Comment by hrubymar10** [comment]
Also I asked @ancestral and now I am asking also @CelticMinstrel and @mattsc , can you provide OpenSSL 1.1 in next mac compile stuff release? This Xcode project depends on it.

Optionaly I can utilize and push my `fix_dependencies.sh` script which uses Homebrew to obtain required Wesnoth's dependencies.

File: projectfiles/Xcode/Wesnoth.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -6175,21 +6241,12 @@
 				);
 				INFOPLIST_FILE = Info.plist;
 				INSTALL_PATH = "$(HOME)/Applications";
-				LIBRARY_SEARCH_PATHS = (
```
Comment (CelticMinstrel) [inline projectfiles/Xcode/Wesnoth.xcodeproj/project.pbxproj]: This looks a little suspicious. What did you change that could've deleted this? If it's just removal of an override setting for a specific target (with the resultant setting remaining the same), then that's fine...

File: projectfiles/Xcode/Wesnoth.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -6175,21 +6241,12 @@
...
-					"$(inherited)",
-					"\"$(SRCROOT)/lib\"",
-				);
 				OTHER_CPLUSPLUSFLAGS = (
 					"$(OTHER_CFLAGS)",
 					"-include",
 					"../../src/boost-patched/bind/arg.hpp",
 					"-I/usr/local/opt/openssl@1.1/include",
 				);
-				"OTHER_CPLUSPLUSFLAGS[arch=*]" = (
```
Comment (CelticMinstrel) [inline projectfiles/Xcode/Wesnoth.xcodeproj/project.pbxproj]: This too looks suspicious; see my previous comment.

**Comment by ancestral** [comment]
I am more than willing to update the Mac Compile Stuff. It is not officially supported, but we like to provide it as a helpful resource to make it easier  to get started building.

It is not a good idea to re-distribute other packages (as licensing restrictions can become a burden), so for that reason, this is a separately maintained project.

If we are wanting libcrypto built from OpenSSL 1.1, I’m willing to update that.

File: projectfiles/Xcode/Wesnoth.xcodeproj/project.pbxproj (hunk 1 above)
Comment (hrubymar10) [reply projectfiles/Xcode/Wesnoth.xcodeproj/project.pbxproj]: This is duplicate override of same thing. Library search paths are defined globaly for whole project on line 6335 (DEBUG) and 6387 (RELEASE). This overrides it with same value so it is useless.

File: projectfiles/Xcode/Wesnoth.xcodeproj/project.pbxproj (hunk 2 above)
Comment (hrubymar10) [reply projectfiles/Xcode/Wesnoth.xcodeproj/project.pbxproj]: This is override of override for specific architecture, which is undefined so every (`*`) and doesn't include `"-I/usr/local/opt/openssl@1.1/include",` . Normal override is on line 6244. This one is useless.

**Comment by hrubymar10** [comment]
I am sorry for delay with answering review requests. I forgot save my comments...

File: projectfiles/Xcode/Wesnoth.xcodeproj/project.pbxproj (hunk 2 above)
Comment (GregoryLundberg) [reply projectfiles/Xcode/Wesnoth.xcodeproj/project.pbxproj]: I think what he's referring to (or, at least, what makes me suspicious) is deleting a left-parenthesis `(` without also deleting a right `)`

Comment (hrubymar10) [reply projectfiles/Xcode/Wesnoth.xcodeproj/project.pbxproj]: You must look at full change. @CelticMinstrel commented just first line of change, but it is longer:
 
image

Comment (GregoryLundberg) [reply projectfiles/Xcode/Wesnoth.xcodeproj/project.pbxproj]: Ah ok

Comment (hrubymar10) [reply projectfiles/Xcode/Wesnoth.xcodeproj/project.pbxproj]: Well it should, this remained from someone and I forgot change it, I'll change it now.

Comment (hrubymar10) [reply projectfiles/Xcode/Wesnoth.xcodeproj/project.pbxproj]: Fixed by https://github.com/wesnoth/wesnoth/pull/2450/commits/721f6d88f71c695db0f1072050bcc753cb6dff78

**Comment by hrubymar10** [comment]
Last commit https://github.com/hrubymar10/wesnoth/commit/100908769ae494cc5f275f7087d4ce4b3a5e4e9d added `Fix_Xcode_Dependencies` script which uses Homebrew (https://brew.sh/) to obtain all required dependencies for Wesnoth

Another change is that Xcode now produces fully working .`app`. Before `.app` depended on Mac Compile Stuff.

Last change is libreadline7.0 requirement

**Comment by hrubymar10** [comment]
Commit https://github.com/wesnoth/wesnoth/pull/2450/commits/2d1d57bce4090a504cec5e3ba110e77475ddcf4c brings some fixes to both scripts like that `cp -rf` != `cp -Rf` and one typo. I tested that after `brew remove boost` Wesnoth.app . Tonight I'll try it also inside clean instalation of macOS 10.13 in VM but I believe that this way works and can produce directly Release `Wesnoth.app` which can be signed and used as official download.

**Comment by hrubymar10** [comment]
Commit https://github.com/wesnoth/wesnoth/pull/2450/commits/6e24eb3e58a25588192e35a732d9209e7724906e is first fully working version of Fix_Xcode_Dependencies script. I tested Wesnoth.app inside clean macOS 10.13.3 install and it worked:
image

**Comment by singalen** [comment]
It's probably good to squash the commits into one.

**Comment by hrubymar10** [comment]
PRs can be merged using `Squash and Merge` button

**Pull request merged**